src/rt/GEAR/rt_thermochemistry_utils.h

```diff
@@ -113,33 +113,52 @@ __attribute__((always_inline)) INLINE static void
 rt_tchem_get_species_densities(const struct part* restrict p, gr_float rho,
                                gr_float species_densities[RT_N_SPECIES]) {
   /* for primordial_chemistry >= 1 */
-  #if GEARRT_GRACKLE_MODE >= 1
-    species_densities[0] = p->rt_data.tchem.mass_fraction_HI * rho;
-    species_densities[1] = p->rt_data.tchem.mass_fraction_HII * rho;
-    species_densities[2] = p->rt_data.tchem.mass_fraction_HeI * rho;
-    species_densities[3] = p->rt_data.tchem.mass_fraction_HeII * rho;
-    species_densities[4] = p->rt_data.tchem.mass_fraction_HeIII * rho;
+  //#if GEARRT_GRACKLE_MODE >= 1
+    //species_densities[0] = p->rt_data.tchem.mass_fraction_HI * rho;
+    //species_densities[1] = p->rt_data.tchem.mass_fraction_HII * rho;
+    //species_densities[2] = p->rt_data.tchem.mass_fraction_HeI * rho;
+    //species_densities[3] = p->rt_data.tchem.mass_fraction_HeII * rho;
+    //species_densities[4] = p->rt_data.tchem.mass_fraction_HeIII * rho;
 
     /* nHII = rho_HII / m_p
      * nHeII = rho_HeII / 4 m_p
      * nHeIII = rho_HeIII / 4 m_p
      * ne = nHII + nHeII + 2 * nHeIII
      * But: it is grackle convention to use rho_e = n_e * m_p */
-    const gr_float rho_e = species_densities[1] + 0.25 * species_densities[3] +
-                         0.5 * species_densities[4];
-    species_densities[5] = rho_e;
-  #endif
+    //const gr_float rho_e = species_densities[1] + 0.25 * species_densities[3] +
+     //                    0.5 * species_densities[4];
+    //species_densities[5] = rho_e;
+  //#endif
 
   /* for primordial_chemistry >= 2 */
-  #if GEARRT_GRACKLE_MODE >= 2
-    species_densities[6] = p->rt_data.tchem.mass_fraction_HM * rho;
-    species_densities[7] = p->rt_data.tchem.mass_fraction_H2I * rho;
-    species_densities[8] = p->rt_data.tchem.mass_fraction_H2II * rho;
+  //#if GEARRT_GRACKLE_MODE >= 2
+    //species_densities[6] = p->rt_data.tchem.mass_fraction_HM * rho;
+    //species_densities[7] = p->rt_data.tchem.mass_fraction_H2I * rho;
+    //species_densities[8] = p->rt_data.tchem.mass_fraction_H2II * rho;
+   
+    //const gr_float rho_e2 = species_densities[1] + 0.25 * species_densities[3] +
+    //                     0.5 * species_densities[4] - species_densities[6] + 0.5 * species_densities[8];
+    //species_densities[5] = rho_e2;
+  //#endif
+  for (enum rt_species species = 0; species < rt_species_count; species++){
+	if (species != 5)
```

Comment:
you shouldn't be comparing to an int, you should be comparing to an rt_species enum, in this case rt_species_e.

However, this way you're just skipping the copying of the electron density. If that doesn't get accessed in the resulting `species_densities` array, then it doesn't hurt to copy it anyway. Performance-wise, an `if` branch within a for loop hurts more than copying over a float.

Alternatively, we could consider moving the `rt_species_e` to be the second-to-last element in the enum. Then you could just loop until rt_species_e, or rt_species_count - 1